deafrica/monitoring/latency_check.py

```diff
@@ -78,19 +111,31 @@ def latency_checker(
             ds = dc.find_datasets(product=satellite, **query)
             print("Datasets since ", date_n_days_ago, " : ", len(ds))
 
-            if len(ds) <= 0:
-                latency_check_slack(
-                    sensor=satellite, notification_url=notification_slack_url
-                )
-            else:
-                print("Latency on ", satellite, " valid.")
-            return 0
+        if len(ds) <= 0 and s3_latency is not None and s3_latency > latency:
+            # Latency exceeded in both Data Cube and S3 bucket
+            latency_check_slack(
+                sensor=satellite,
+                exceeded="Latency exceeded in Data Cube and S3 bucket",
+                notification_url=notification_slack_url,
+            )
+        elif len(ds) <= 0:
+            # Latency exceeded in Data Cube
+            latency_check_slack(
+                sensor=satellite,
+                exceeded="Latency exceeded in Data Cube",
+                notification_url=notification_slack_url,
+            )
+        elif s3_latency is not None and s3_latency > latency:
+            # Latency exceeded in S3 bucket
+            latency_check_slack(
+                sensor=satellite,
+                exceeded="Latency exceeded in S3 bucket",
+                notification_url=notification_slack_url,
+            )
         else:
-            print("Invalid Product!")

```

Comment:
Any specific reason for removing these invalid latency/satellite printouts?

Reply:
I didn't have any specific reason to remove them to be honest. I just thought they were no longer neccesary. However, I'll add them back. I only focused on cases where the product is valid. I hope that makes sense.